contracts/core/Registry.sol

```diff
@@ -21,27 +36,75 @@ contract Registry {
     // attestationAddr = addr(hash(id, name, owner))
     // OR deploy right away with CREATE2 as proxy so they can do anything (or just forward ERC20s and ETH)
     // id => IdentityDetails
-    function createIdentity(IdentityDetails memory _identityDetails, address[] memory _owners) external returns (uint256) {
 
+    // ASK: changing the function signature as attestationAddr has to be generated from the name and caller
+    /// @notice Creates a new identity
+    /// @dev This will also set the attestation address generated from msg.sender and name
+    function createIdentity(
+        string memory name,
+        Metadata memory metadata,
+        address[] memory _owners
+    ) external returns (uint256) {
+
+        IdentityDetails memory identityDetails = IdentityDetails(
+            name,
+            metadata,
+            // ASK : should identityId be used as the hash for attestationAddr?
+            // cause project can't have same attestAddr across chains
+            generateAttestationAddr(name, msg.sender)
+        );
+
+        identities[identityId++] = identityDetails;
+
+        for(uint i = 0; i < _owners.length; i++) {
+            _grantRoles(_owners[i], identityId);
```

Comment:
identityId is being used as the Role here.

Reply:
We want to add an extra variable because there are two types of roles, owner and member. 

Owners can change the metadata.

Members can claim to be a part of the organization when creating a pool or applying.

I would create an enum for types of roles (OWNER and MEMBER) and then would `abi.encode(identityId, roleType)` to get the role for Solady.

Reply:
Makes sense 🙌🏼